clash-protocols/src/Protocols/Wishbone/Standard/Hedgehog.hs

```diff
@@ -441,6 +473,30 @@ validatorCircuitLenient =
       Right (True, state1) -> ((cycle + 1, (m2s1, s2m1), state1), (s2m1, m2s1))
       Right (False, state1) -> go (cycle, (m2s0, s2m0), state1) (m2s1, s2m1)
 
+{- |
+  Takes elements from a list while the predicate holds, considers up to @window@ elements
+  since the last element that satisfied the predicate.
+
+  >>> takeWhileSparse 3 Prelude.odd [1, 2, 3, 6, 8, 10, 12]
+  [1,2,3]
```

Comment:
```suggestion
Takes elements from a list while the predicate holds, considers up to @window@ elements
since the last element that satisfied the predicate.

>>> takeWhileSparse 3 Prelude.odd [1, 2, 3, 6, 8, 10, 12]
[1,2,3]
```